emmet-core/emmet/core/mpid.py

```diff
@@ -0,0 +1,24 @@
+from dataclasses import dataclass
+from typing import Union
+
+NOTHING = object()
+
+
+class MPID(str):
```

Comment:
Usually wary of subclassing builtin types, but usage seems appropriate.

Reply:
Yeah, I was super wary, but this is one of the few situations where it makes sense.